src/htcondor_es/convert_to_json.py

```diff
@@ -559,11 +564,13 @@ def get_creation_time_from_taskname(ad):
 _generic_site = re.compile("^[A-Za-z0-9]+_[A-Za-z0-9]+_(.*)_")
 _cms_site = re.compile("CMS[A-Za-z]*_(.*)_")
 _cmssw_version = re.compile("CMSSW_((\d*)_(\d*)_.*)")
-def convert_to_json(ad, cms=True, return_dict=False, reduce_data=False):
+
+def convert_to_json(ad, cms=True, return_dict=False, reduce_data=False, aff_mgr=None):
     if ad.get("TaskType") == "ROOT":
         return None
+    if not aff_mgr:
```

Comment:
I would keep the option to call `convert_to_json` without this feature. So rather than creating a new AffiliationManager when it hasn't been passed, check whether it has been passed and then only query it in that case and leave the affiliation info unfilled if it hasn't.